ycms/cms/models/room.py

```diff
@@ -99,7 +102,61 @@ def genders(self):
         :return: genders of patients in the room
         :rtype: set
         """
-        return set(patient.gender for patient in self.patients.all())
+        return set(patient.gender for patient in self.patients())
+
+    @cached_property
+    def insurance_types(self):
+        """
+        Helper property for accessing all insurance_types of patients currently stationed in the room
+
+        :return: insurance_types of patients in the room
+        :rtype: set
+        """
+        return set(patient.insurance_type for patient in self.patients())
+
+    @cached_property
+    def patient_ages(self):
+        """
+        Helper property for accessing ages of patients currently stationed in the room
+
+        :return: ages of patients in the room
+        :rtype: list
+        """
+        return [patient.age for patient in self.patients()]
```

Comment:
How about this:
```suggestion
        return [patient.age for patient in self.patients()] if self.patients() else [0]
```
so we can then have an easier time with the other properties (see next comments)?